src/nm.c

```diff
@@ -545,6 +545,11 @@ write_fallback_key_value(GQuark key_id, gpointer value, gpointer user_data)
         group = g_strjoinv(".", group_key); //re-combine group parts
     }
 
+    if (!g_strcmp0(group, "")) {
+        g_warning("NetworkManager: keyfile group is empty for key '%s'.", key);
+        goto cleanup;
+    }
```

Comment:
The assertion seems to stem from GLib's `g_key_file_is_group_name()` check: https://github.com/GNOME/glib/blob/main/glib/gkeyfile.c#L4158

But we're already in the output stage here in `nm.c`. The data that we have in our datastructures should be valid. So this seems more like a workaround. I feel like this should rather be fixed in `parse-nm.c` to avoid getting an invalid `group.key` value into the datastructures in the first place.

Reply:
In this case we actually read the passthrough data in handle_generic_datalist(). It doesn't do any parsing. If we want to not load invalid configuration I guess that would be the place.

Reply:
But well, handle_generic_datalist() is generic so it doesn't need to know about the structure of the data it's reading...